api/api_test.go

```diff
@@ -115,6 +119,53 @@ func TestServe_context_cancel(t *testing.T) {
 	}
 }
 
+func TestServeWithTLS(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	port := testutils.FreePort(t)
+
+	task, err := driver.NewTask(driver.TaskConfig{Enabled: true})
+	require.NoError(t, err)
+
+	drivers := driver.NewDrivers()
+	d := new(mocks.Driver)
+	d.On("UpdateTask", mock.Anything, mock.Anything).
+		Return(driver.InspectPlan{}, nil).Once()
+	d.On("Task").Return(task)
+	drivers.Add("task_b", d)
+
+	cert := "../testutils/localhost_cert.pem"
+	tlsConfig := &config.TLSConfig{
+		Enabled: config.Bool(true),
+		Cert:    config.String(cert),
+		Key:     config.String("../testutils/localhost_key.pem"),
+	}
+	api := NewAPI(event.NewStore(), drivers, port, tlsConfig)
+	go api.Serve(ctx)
+	time.Sleep(3 * time.Second)
+
+	// Set up a client that trusts the self-signed certificate
+	caCert, err := ioutil.ReadFile(cert)
+	require.NoError(t, err)
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: caCertPool,
+		},
+	}}
+
+	// Make request to HTTPS endpoint
+	u := fmt.Sprintf("https://localhost:%d/%s/status",
+		port, defaultAPIVersion)
+	resp, err := client.Get(u)
```

Comment:
Can you add coverage to test a request from a client w/o certs to a sever with it enabled?